pkg/reconciler/ingress/config/istio.go

```diff
@@ -106,9 +106,14 @@ func parseGateways(configMap *corev1.ConfigMap, prefix string) ([]Gateway, error
 	sort.Strings(gatewayNames)
 	gateways := make([]Gateway, len(gatewayNames))
 	for i, gatewayName := range gatewayNames {
+		namespace := system.Namespace()
+		idx := strings.Index(gatewayName, ".")
+		if idx != -1 {
+			namespace = gatewayName[:idx]
```

Comment:
I think the code is correct, but the index operations are kinda hard to dechiffre. Should we maybe do a `strings.SplitN` operation on the gatewayName and then handle cases for `len(parts) == 1` (system namespace), `2` (namespace given)

Reply:
Switched to strings.SplitN, which is indeed easier to understand. About the readability, do you think the variable name **gatewayName** is a bit confusing as it may contain both namespace and name? Thank you.